op-service/txmgr/cli.go

```diff
@@ -0,0 +1,175 @@
+	ResubmissionTimeoutFlagName       = "resubmission-timeout"
+)
+
+func CLIFlags(envPrefix string) []cli.Flag {
+	return append([]cli.Flag{
+		cli.StringFlag{
+			Name:   MnemonicFlagName,
+			Usage:  "The mnemonic used to derive the wallets for either the service",
+			EnvVar: opservice.PrefixEnvVar(envPrefix, "MNEMONIC"),
+		},
+		cli.StringFlag{
+			Name:   HDPathFlagName,
+			Usage:  "The HD path used to derive the sequencer wallet from the mnemonic. The mnemonic flag must also be set.",
+			EnvVar: opservice.PrefixEnvVar(envPrefix, "HD_PATH"),
+		},
+		cli.StringFlag{
+			Name:   "private-key",
+			Usage:  "The private key to use with the service. Must not be used with mnemonic.",
+			EnvVar: opservice.PrefixEnvVar(envPrefix, "PRIVATE_KEY"),
+		},
+		cli.Uint64Flag{
+			Name:   NumConfirmationsFlagName,
+			Usage:  "Number of confirmations which we will wait after sending a transaction",
+			Value:  10,
+			EnvVar: opservice.PrefixEnvVar(envPrefix, "NUM_CONFIRMATIONS"),
+		},
+		cli.Uint64Flag{
+			Name:   "safe-abort-nonce-too-low-count",
+			Usage:  "Number of ErrNonceTooLow observations required to give up on a tx at a particular nonce without receiving confirmation",
+			Value:  3,
+			EnvVar: opservice.PrefixEnvVar(envPrefix, "SAFE_ABORT_NONCE_TOO_LOW_COUNT"),
+		},
+		cli.DurationFlag{
+			Name:   "resubmission-timeout",
+			Usage:  "Duration we will wait before resubmitting a transaction to L1",
+			Value:  30 * time.Second,
+			EnvVar: opservice.PrefixEnvVar(envPrefix, "RESUBMISSION_TIMEOUT"),
+		},
+	}, client.CLIFlags(envPrefix)...)
+}
+
+type CLIConfig struct {
+	L1RPCURL                  string
+	Mnemonic                  string
+	SequencerHDPath           string
+	PrivateKey                string
+	SignerCLIConfig           client.CLIConfig
+	NumConfirmations          uint64
+	SafeAbortNonceTooLowCount uint64
+	ResubmissionTimeout       time.Duration
+}
+
+func (m CLIConfig) Check() error {
+	if m.L1RPCURL == "" {
+		return errors.New("must provide a L1 RPC url")
+	}
+	if err := m.SignerCLIConfig.Check(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func ReadCLIConfig(ctx *cli.Context) CLIConfig {
+	return CLIConfig{
+		L1RPCURL:                  ctx.GlobalString(L1RPCFlagName),
+		Mnemonic:                  ctx.GlobalString(MnemonicFlagName),
+		SequencerHDPath:           ctx.GlobalString(HDPathFlagName),
+		PrivateKey:                ctx.GlobalString(PrivateKeyFlagName),
+		SignerCLIConfig:           client.ReadCLIConfig(ctx),
+		NumConfirmations:          ctx.GlobalUint64(NumConfirmationsFlagName),
+		SafeAbortNonceTooLowCount: ctx.GlobalUint64(SafeAbortNonceTooLowCountFlagName),
+		ResubmissionTimeout:       ctx.GlobalDuration(ResubmissionTimeoutFlagName),
+	}
+}
+
+func NewConfig(cfg CLIConfig, l log.Logger) (Config, error) {
+	if err := cfg.Check(); err != nil {
+		return Config{}, err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
```

Comment:
```suggestion
	networkTimeout := 2 * time.Second

	ctx, cancel := context.WithTimeout(context.Background(), networkTimeout)
```
I'd set once and then reuse (3 times). 

When it is added to the `CLIConfig` in a follow-up, can then just be read from there.